cEP-0020.md

```diff
@@ -0,0 +1,355 @@
+
+**The problems we have:**
+
+- How to attract more newcomers
+  - How to cope with the increasing amounts of newcomers
+- How to teach and help newcomers to learn faster so
+  we can keep working on coala at a high pace.
+- How to improve newcomer -> dev/maintainer ratio
+- How to increase amount of feedback we get
+  - How to use feedback to improve coala and processes
+
+## Why Gamification?
+
+As coala tries to be a welcoming organization for newcomers, it gives them
+clear pathways and as much direct assistance as possible. However, it is
+already known that newcomers face many barriers while attempting to contribute
+to the open source community for the first time. Some barriers they face include
+orientation issues that can potentially demotivate newcomers from placing
+their first contribution. On the other hand, gamification is widely used to
+engage and motivate people to accomplish tasks and improve their performance.
+Therefore, the goal of this project is to use gamification to orient and
+motivate newcomers to overcome onboarding barriers to contribute and engage
+with coala and its community.
+
+## Gamification Elements
+
+There will be three game design elements to help newcomers: Points, Levels
+and Badges. For each of these elements, we will have a set of rules to
+describe the operation of the gamified environment. i.e.:
+
+### Points and Rules
+
+Let’s start by defining some rules for earning points. Points are a
+simple tool to reward behavior, and they provide an excellent base upon which to
+build a gamified design. The first step is to define the activities that we want
+newcomers to take. i.e.:
+
+- Created Issues by a Newcomer
+- Created Pull Requests by a Newcomer
+- Number of Gitmate errors on the PR
+- CI status of the PR
+- Number of reviews done by a Newcomer
+- Number of comments in one review by a Newcomer
+- Comments on the Issues by a Newcomer
+- Activities on Gitter channels
+- Introduce `.coafile` in other projects
+
+Then we will assign some points to each of the activity a newcomer perform.
+E.g.: If a user is creating a difficulty/newcomer issue he will be awarded
+with `10` points, similarly, the value of the points will be increased
+according to the difficulty level of the issue.
+
+### Levels
+
+To increase a player’s competitive instinct, we can implement a series of
+levels that confer rank as newcomers become more active. A good and
+straightforward way to award levels in a new gamified design is to base them
+on point thresholds. As players earn points, they move up an incremental
+series of levels. To infuse the level with meaning i.e.: a ranking within the
+system, the levels should be named in a way that indicates status.
+
+- Level-I:
+  - Name: Beginner
+  - Points_required = 50
+- Level-II:
+  - Name: Intermideate
+  - Points_required = 150
+- Level-III:
+  - Name: High
+  - Points_required = 300
+
+More complex designs, such as awarding levels based on consecutive activities
```

Comment:
What privileges could we give to them in each of the levels?
- The one I could think of is to give them permission to use corobo after the completion of level one?
Also, it would be good to have some privileges after a newcomer has earned the `coala Developer` badge.

Reply:
Another one suggested by @ishanSrt:
- Give permission to ping maintainers on gitter after some points threshold.